fxa-auth-db-server/test/backend/remote.js

```diff
@@ -594,14 +607,89 @@ module.exports = function(cfg, server) {
+          t.equal(token.tokenVerificationId, user.keyFetchToken.tokenVerificationId, 'tokenVerificationId is correct')
+
+          // Attempt to verify a non-existent key fetch token
+          return client.postThen('/tokens/' + crypto.randomBytes(16).toString('hex') + '/verify', {
+            uid: user.accountId
+          })
+        })
+        .then(function(r) {
+          t.fail('Verifying a non-existent token should fail')
+        }, function(err) {
+          testNotFound(t, err)
+
+          // Attempt to verify a key fetch token with the wrong uid
+          return client.postThen('/tokens/' + user.keyFetchToken.tokenVerificationId + '/verify', {
+            uid: crypto.randomBytes(16).toString('hex')
+          })
+        })
+        .then(function(r) {
+          t.fail('Verifying a non-existent token should fail')
+        }, function(err) {
+          testNotFound(t, err)
+
+          // Verify the key fetch token
+          return client.postThen('/tokens/' + user.keyFetchToken.tokenVerificationId + '/verify', {
+            uid: user.accountId
+          })
+        })
+        .then(function() {
+          // Fetch the key fetch token
+          return client.getThen('/keyFetchToken/' + user.keyFetchTokenId)
+        })
+        .then(function(r) {
+          t.equal(r.obj.tokenVerificationId, undefined, 'tokenVerificationId is undefined')
+
+          // Fetch the key fetch token with its verification state
+          return client.getThen('/keyFetchToken/' + user.keyFetchTokenId + '/verified')
+        })
+        .then(function(r) {
+          t.equal(r.obj.tokenVerificationId, null, 'tokenVerificationId is null')
+
+          // Create a verified key fetch token
+          return client.putThen('/keyFetchToken/' + verifiedUser.keyFetchTokenId, verifiedUser.keyFetchToken)
+        })
+        .then(function () {
+          // Fetch the verified key fetch token
+          return client.getThen('/keyFetchToken/' + verifiedUser.keyFetchTokenId)
+        })
+        .then(function(r) {
+          t.equal(r.obj.tokenVerificationId, undefined, 'tokenVerificationId is undefined')
+
+          // Fetch the verified key fetch token with its verification state
+          return client.getThen('/keyFetchToken/' + verifiedUser.keyFetchTokenId + '/verified')
+        })
+        .then(function(r) {
+          t.equal(r.obj.tokenVerificationId, null, 'tokenVerificationId is null')
+
+          // Delete both key fetch tokens
+          return P.all([
+            client.delThen('/keyFetchToken/' + user.keyFetchTokenId),
+            client.delThen('/keyFetchToken/' + verifiedUser.keyFetchTokenId)
+          ])
+        })
+        .then(function(results) {
+          t.equal(results.length, 2)
+          results.forEach(function (result) {
+            respOk(t, result)
+          })
 
-          // Attempt to fetch the deleted key fetch token
+          // Attempt to fetch a deleted key fetch token
```

Comment:
It is not entirely obvious if this test case has been covered. Creating a unverified KeyFetchToken, deleting token then request to verify token (`/verifyTokens`) returns 404. Basically a test case to make sure that deleting a KeyFetchToken will remove tokens in the unverified table as well.